src/db/homeseeker/property.ts

```diff
@@ -0,0 +1,189 @@
+): Promise<Property[]> {
+	const [res] = await pool.execute<Property[]>(
+		`SELECT * FROM bookings_db.hs_property
+        WHERE zipcode = :zipcode`,
+		{ zipcode },
+	);
+	return res;
+}
+
+/**
+ * Retrieve a property with given id
+ */
+export async function getPropertyByID(id: number): Promise<Property | null> {
+	const [res] = await pool.execute<Property[]>(
+		`SELECT * FROM bookings_db.hs_property
+        WHERE id = :id`,
+		{ id },
+	);
+	if (res.length !== 1) {
+		return null;
+	}
+	return res[0];
+}
+
+/**
+ * Retrieve a property with given address
+ */
+export async function getPropertyByAddress(
+	address: string,
+): Promise<Property | null> {
+	const [res] = await pool.execute<Property[]>(
+		`SELECT * FROM bookings_db.hs_property
+        WHERE address = :address`,
+		{ address },
+	);
+	if (res.length === 1) {
+		return res[0];
+	}
+	return null;
+}
+
+/**
+ * Retrieve a property with given schedule
+ */
+export async function getPropertyBySchedule(
+	schedule_id: number,
+): Promise<Property | null> {
+	const [res] = await pool.execute<Property[]>(
+		`SELECT *
+		FROM bookings_db.hs_property AS P
+		INNER JOIN bookings_db.hs_schedule as S
+        ON S.property_id = P.id
+		WHERE S.id = :schedule_id`,
+		{ schedule_id },
+	);
+	if (res.length === 1) {
+		return res[0];
+	}
+	return null;
+}
+
+/**
+ * Creates a property with the given parameters.
+ *
+ * @returns id of newly created property
+ */
+
+export async function createProperty(
+	broker_id: number,
+	address: string,
+	zipcode: number,
+	type: string,
+	price?: number,
+	rooms?: number,
+	area?: number,
+	built?: number,
+): Promise<number | null> {
+	try {
+		const params = {
+			broker_id,
+			address,
+			zipcode,
+			type,
+			price: price !== undefined ? price : null,
+			rooms: rooms !== undefined ? rooms : null,
+			area: area !== undefined ? area : null,
+			built: built !== undefined ? built : null,
+		};
+
+		const [res] = await pool.execute<ResultSetHeader>(
+			`INSERT INTO bookings_db.hs_property (broker_id, address, zipcode, type, price, rooms, area, built)
+            VALUES (:broker_id, :address, :zipcode, :type, :price, :rooms, :area, :built)`,
+			params,
+		);
+		return res.insertId;
+	} catch (e) {
+		console.error(e);
+		return null;
+	}
+}
+
+/**
+ * Deletes a property with given id.
+ */
+export async function deleteProperty(id: number): Promise<void> {
+	await pool.execute("DELETE FROM bookings_db.hs_property WHERE id = :id", {
+		id,
+	});
+}
+
+/**
+ * Update certain fields of a property with a given id.
+ */
+export async function updateProperty(
+	id: number,
+	address?: string,
+	zipcode?: number,
+	type?: string,
+	price?: number,
+	rooms?: number,
+	area?: number,
+	built?: number,
+): Promise<void> {
+	const params = {
+		address,
+		zipcode,
+		type,
+		price: price !== undefined ? price : null,
+		rooms: rooms !== undefined ? rooms : null,
+		area: area !== undefined ? area : null,
+		built: built !== undefined ? built : null,
```

Comment:
Is the intention with these to make the function arguments optional? Or is it so you can explicitly pass in `undefined` or `null` to clear the column?

Reply:
I think the intention was for it to be optional parameter and in SQL they accept null and not undefined but in the page if someone didn't fill some part of the form it is undefined. So this basically make the SQL query have the correct type when updating/inserting.